source/fundamentals/odata.txt

```diff
@@ -0,0 +1,203 @@
+      .. literalinclude:: /includes/code-examples/GradeEntry.cs
+         :language: csharp
+         :copyable:
+         :dedent:
+
+      .. include:: /includes/convention-pack-note.rst
+
+   .. step:: Create an OData Controller
+
+      Create a new folder in your solution called ``Controllers`` and add a new
+      controller file called ``RestaurantsController.cs``. Copy the following code
+      into the file:
+
+      .. literalinclude:: /includes/fundamentals/code-examples/connection/OData.cs
+         :language: csharp
+         :start-after: // start-controller
+         :end-before: // end-controller
+
+      This code performs the following actions:
+
+      - Creates a constructor that connects to MongoDB, and gets the
+        ``restaurants`` collection.
+      - Creates a ``Get`` endpoint that returns all restaurants in the collection.
+      - Specifies the ``MongoEnableQuery`` attribute to enable querying on the
+        ``Get`` endpoint.
+      - Specifies the ``PageSize`` attribute on ``MongoEnableQuery`` to limit the
+        number of documents returned to ``5``.
+
+   .. step:: Configure the OData Service
+
+      Paste the following code into your ``Program.cs`` file to configure the OData
+      service and map your controller endpoints.
+
+      .. literalinclude:: /includes/fundamentals/code-examples/connection/OData.cs
+         :language: csharp
+         :start-after: // start-configure
+         :end-before: // end-configure
+
+      .. note::
+
+         Replace the ``<"Your connection URI">`` placeholder with your MongoDB connection string.
+      
+      This code performs the following actions:
+
+      - Instantiates a new ``MongoClient`` and passes it to the controller
+      - Defines the Entity Data Model (EDM) and registers ``Restaurants`` as an
+        entity set with the key, ``Id``.
+      - Adds the OData service and enables the ``Select()`` query capability.
+      - Registers the route by using the ``AddRouteComponents()`` method.
+      - Calls the ``UseRouting()`` and ``MapControllers()`` methods to match
+        incoming HTTP requests and route them to the appropriate endpoint.
+
+   .. step:: Run the Application
+
+      Run the application by using your IDE, or by running the following command
+      in your shell at the root directory of your project:
+
+      .. code-block:: shell
+
+         dotnet run ODataExample.csproj
+
+      After running the application, your terminal displays output similar
+      to the following:
+
+      .. code-block:: none
+
+         info: Microsoft.Hosting.Lifetime[14]
+               Now listening on: http://localhost:5183
+         info: Microsoft.Hosting.Lifetime[0]
+               Application started. Press Ctrl+C to shut down.
+         info: Microsoft.Hosting.Lifetime[0]
+               Hosting environment: Development
+         info: Microsoft.Hosting.Lifetime[0]
+               Content root path: /Users/jordan.smith/Documents/mongo/ODataTest/ODataTest
```

Comment:
[q] Should this be an [example name](https://www.mongodb.com/docs/meta/style-guide/style/names/) or do you not mind having your name here? 😄

Reply:
Good call, updated to be a placeholder instead!